modules/adb.nix

```diff
@@ -0,0 +1,51 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+with import ./initrd-order.nix;
+
+{
+  options.mobile.adbd = {
+    enable = mkOption {
+      type = types.bool;
+      default = config.mobile.system.type == "android";
+      description = ''
+        Enables adbd on the device.
+      '';
+    };
+  };
+
+  config = lib.mkIf config.mobile.adbd.enable {
+    assertions = [
+      { assertion = config.mobile.system.type == "android";
+        message = "adb is only available on Android";
+      }
+      { assertion = config.mobile.boot.stage-1.usb.enable;
+        message = "adb requires mobile.boot.stage-1.usb.enable = true";
+      }
+    ];
+
+    mobile.boot.stage-1 = {
+      usb.features = [ "ffs" ];
+
+      init = lib.mkOrder AFTER_DEVICE_INIT ''
+        (
+        mkdir -p /dev/usb-ffs/adb
+        mount -t functionfs adb /dev/usb-ffs/adb/
```

Comment:
Playing with WIP and bleeding edge code is always fun :sweat_drops: 

With the lavender port, I [had to implement the new gadgetfs stuff](https://github.com/samueldr/mobile-nixos/pull/28/commits/9e0e65c09a544a7d2b9d30e0901d925f3ffeeded), and my experience with it is that the functionfs stuff is actually related to it... When I'll be back home, with devices ready to test, I'll figure out what's actually required to happen and in what order for those devices.

Though, if I remember correctly, with the pixel 2 I had kernel panics when doing things in the wrong order, but it might not be functionfs-related, fuzzy about the details, and don't have one of those handy currently.